lms/djangoapps/commerce/utils.py

```diff
@@ -274,16 +281,111 @@ def refund_seat(course_enrollment, change_mode=False):
-                                                is_active=False, skip_refund=True)
-            course_enrollment.save()
+        if change_mode:
+            _auto_enroll(course_enrollment)
     else:
         log.info('No refund opened for user [%s], course [%s]', enrollee.id, course_key_str)
 
     return refund_ids
 
 
+def _refund_in_commerce_coordinator(course_enrollment, change_mode):
+    """
+    Helper function to perform refund in Commerce Coordinator.
+
+    Parameters:
+        course_enrollment (CourseEnrollment): the enrollment to refund.
+        change_mode (bool): whether the enrollment should be auto-enrolled into
+            the default course mode after refund.
+
+    Returns:
+        bool: True if refund was performed. False if refund is not applicable
+            to Commerce Coordinator.
+    """
+    enrollment_source_system = course_enrollment.get_order_attribute_value("source_system")
+    course_key_str = str(course_enrollment.course_id)
+
+    # Commerce Coordinator enrollments will have an orders.source_system enrollment attribute.
+    # Redirect to Coordinator only if the source_system is safelisted as Coordinator's in settings.
+
+    if enrollment_source_system and enrollment_source_system in settings.COMMERCE_COORDINATOR_REFUND_SOURCE_SYSTEMS:
+        log.info('Redirecting refund to Commerce Coordinator for user [%s], course [%s]...',
+                 course_enrollment.user_id, course_key_str)
+
+        # Re-use Ecommerce API client factory to build an API client for Commerce Coordinator...
+        service_user = get_user_model().objects.get(
+            username=settings.COMMERCE_COORDINATOR_SERVICE_WORKER_USERNAME
+        )
+        api_client = get_ecommerce_api_client(service_user)
+        refunds_url = urljoin(
+            settings.COMMERCE_COORDINATOR_URL_ROOT,
+            settings.COMMERCE_COORDINATOR_REFUND_PATH
+        )
+
+        # Build request, raising exception if Coordinator returns non-200.
+        enrollment_attributes = CourseEnrollmentAttribute.get_enrollment_attributes(course_enrollment)
+
+        try:
+            api_client.post(
+                refunds_url,
+                json={
+                    'course_id': course_key_str,
+                    'username': course_enrollment.username,
+                    'enrollment_attributes': enrollment_attributes
+                }
+            ).raise_for_status()
+
+        except Exception as exc:  # pylint: disable=broad-except
+            # Catch any possible exceptions from the Commerce Coordinator service to ensure we fail gracefully
+            log.exception(
+                "Unexpected exception while attempting to refund user in Coordinator [%s], "
+                "course key [%s] message: [%s]",
+                course_enrollment.username,
+                course_key_str,
+                str(exc)
+            )
+            return False
```

Comment:
We can remove this as we don't need to return False here. We are returning a True at the end of this if block.

More context: Returning a False value here means existing code (legacy e-commerce refund) logic gets executed which we don't want to do for purchases with other source systems.